packages/fxa-settings/src/contexts/DynamicLocalizationContext.tsx

```diff
@@ -79,10 +80,51 @@ export const DynamicLocalizationProvider: React.FC<{
     }
   }, [currentLocale]);
 
+  const clearLanguagePreference = useCallback(async () => {
+    setIsLoading(true);
+
+    try {
+      // 1. Clear preference from localStorage
+      clearLocalePreference();
+
+      // 2. Get browser's default locale directly (ignore document.lang)
+      let browserDefaultLocale = 'en'; // fallback
+      
+      // Check navigator.language first
+      const browserLang = navigator.language;
+      if (browserLang && supportedLanguages.includes(browserLang)) {
+        browserDefaultLocale = browserLang;
+      } else {
+        // Check base language (e.g., 'en' from 'en-US')
+        const baseLang = browserLang?.split('-')[0];
+        if (baseLang && supportedLanguages.includes(baseLang)) {
+          browserDefaultLocale = baseLang;
+        }
+      }
+
+      // 3. Update document attributes
+      document.documentElement.lang = browserDefaultLocale;
```

Comment:
Just want to point out that the "proper" way to do this is probably using `react-helmet`. Not sure how much of a refactor that would be to do here though.